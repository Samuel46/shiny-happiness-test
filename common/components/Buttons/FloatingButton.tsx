import React, { ButtonHTMLAttributes, FC } from 'react';

import styles from '../../../styles/Buttons.module.css';

type FloatingButtonProps = ButtonHTMLAttributes<HTMLButtonElement>;

const FloatingButton: FC<FloatingButtonProps> = (props) => (
  <div className={styles.floatingButtonContainer}>
    <button type="button" className={styles.floatingButton} {...props}>
      {props.children}
    </button>
  </div>
);

export { FloatingButton };
