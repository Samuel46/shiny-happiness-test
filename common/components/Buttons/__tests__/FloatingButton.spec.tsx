import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';

import { FloatingButton } from '../FloatingButton';

it('should render', () => {
  const { asFragment } = render(
    <FloatingButton onClick={() => {}}>A Unique Button</FloatingButton>,
  );

  expect(asFragment()).toMatchSnapshot();
});

it('should call the handler on click', () => {
  const onClick = jest.fn();

  render(
    <FloatingButton onClick={onClick}>Another Unique Button</FloatingButton>,
  );

  fireEvent.click(screen.getByRole('button'));

  expect(onClick).toHaveBeenCalled();
});
