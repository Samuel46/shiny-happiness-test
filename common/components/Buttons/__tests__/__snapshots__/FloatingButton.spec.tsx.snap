// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<DocumentFragment>
  <div
    class="floatingButtonContainer"
  >
    <button
      class="floatingButton"
      type="button"
    >
      A Unique Button
    </button>
  </div>
</DocumentFragment>
`;
